src/ekat/ekat_pack_kokkos.hpp

```diff
@@ -249,15 +249,15 @@ repack (const Kokkos::View<Pack<T, old_pack_size>*, Parms...>& vp) {
 //
 
 // 1d
-template <size_t N, typename ViewT>
+template <typename SizeT, size_t N, typename ViewT>
 void host_to_device(const Kokkos::Array<typename ViewT::value_type::scalar const*, N>& data,
-                    const Kokkos::Array<size_t, N>& sizes,
+                    const Kokkos::Array<SizeT, N>& sizes,
                     Kokkos::Array<ViewT, N>& views)
 {
   using PackT = typename ViewT::value_type;
 
   for (size_t i = 0; i < N; ++i) {
-    const size_t size = sizes[i];
+    const size_t size = static_cast<size_t>(sizes[i]);
```

Comment:
Is there really a need to cast to size_t? Why not simply using `sizes[i]`?

Reply:
I don't think there's a need, but implicit casts can be a source of error, so I'm not in the habit of using them outside of C (and even then I usually cast explicitly). Maybe it's superstitious, but C++ has enough bizarre rules that I'm in favor of making things like this more explicit anyway.

Reply:
Ah, true, there would otherwise be an implicit cast when you compute npacks. Then yeah, keep static casts.